src/NServiceBus.Testing/NSB.Testing.Fakes/AsyncLocal.cs

```diff
@@ -0,0 +1,36 @@
+#if NETFRAMEWORK
```

Comment:
I guess we would notice this to be an issue once upgrading the framework to a newer version? Otherwise should we consider targeting the specific framework version on the if?

Reply:
I used the same as in the core introduced in https://github.com/Particular/NServiceBus/pull/5202